SYCL/ESIMD/api/functional/functions/functions_select_rvalue.hpp

```diff
@@ -0,0 +1,204 @@
+//===-- functions_select_rvalue.hpp - Functions for tests on simd rvalue select
+//      function. ---------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd rvalue select function.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "../shared_element.hpp"
+#include "common.hpp"
+
+// for std::ceil
+#include <cmath>
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+
+namespace esimd_test::api::functional::functions {
+
+// Aliases to provide size or stride values to test.
+template <int N> using stride = std::integral_constant<int, N>;
+template <int N> using size = std::integral_constant<int, N>;
```

Comment:
The `_type` prefix has been added [here](https://github.com/intel/llvm-test-suite/pull/875/commits/9519eccf663e28b1760cf5bf869ba75ea5fa275d#diff-ee52ba82babf88cc257b75cc71c61eca3df7aa19a2b6ce7b5ed19349642a56f6R38-R41)

> BTW, why not to use a integer_pack from type_coverage.hpp with for_all_combinations?

Now we have an self-describe variables like `stride_type`, and by using the `integer_pack` we will iterate over combinations of different sizes and strides, but this iteration is useless 

> Also please note that using doesn't provide a new type, so there is no additional type safety in different aliases. So you might pass a size<3> instead of stride<3> with no errors.

No we don't have a good solution to solve this problem